operator/pkg/manifest/helm_client_cache.go

```diff
@@ -36,10 +39,29 @@ func (h *HelmClientCache) Get(key client.ObjectKey) types.HelmClient {
 
 // Set saves the passed HelmClient into HelmClientCache for the client.ObjectKey.
 func (h *HelmClientCache) Set(key client.ObjectKey, helmClient types.HelmClient) {
-	h.cache.Store(key, helmClient)
+	h.clientCache.Store(key, helmClient)
```

Comment:
you are storing types.HelmClient but are loading CachedDiscoveryClient here. Is this working as intended?